src/components/query-panel/QueryPanelCtrl.ts

```diff
@@ -0,0 +1,84 @@
+import CodeMirror = require("codemirror");
+import {RestService} from "../../services/rest/RestService";
+import {EditorConfiguration} from "codemirror";
+import {isNotNullOrUndefined, isObject, isEmptyString} from "../../common/utils/Utils";
+
+export class QueryPanelCtrl implements ng.IComponentController {
+  static $inject: string[] = ["restService", "$timeout"];
+
+  cacheName: string;
+  name: string;
+  description: string;
+  statusCode: number;
+  editor: CodeMirror.Editor;
+  panelType:string;
+  inputParams: string [];
+
+  constructor(private restService: RestService, private $timeout: ng.ITimeoutService) {
+  }
+
+  executeAction(): void {
+    this.executionActionHelper(this.name, this.inputParams[0], this.inputParams[1]).then((response: any) => {
```

Comment:
As you are using class variables, there is no need to pass these values to `executionActionHelper`, instead you can just access them in the method.

Reply:
+1, done